### PR TITLE
Model level eager loading configuration

### LINKED ISSUE #21540: Query: Model configuration for eager load (Include)
This is split off from #2953 

This issue tracks about ability to configure a navigation to be eager loaded in model building which will be eager loaded during query without having to use Include.
And additional API `IgnoreEagerLoadedNavigations` on Queryable to ignore model based config and not load eager loaded navigation (except ownership navigations)

### PR DESCRIPTION
**Pull request by smitpatel**
IsEagerLoaded API on NavigationBuilder to mark a navigation as eagar loaded
IgnoreEagerLoadedNavigations API on query to ignore model based config

Resolves #21540

### CODE REVIEW

**Comment by ajcvickers** [comment]
Tests?

File: src/EFCore/Query/Internal/NavigationExpandingExpressionVisitor.cs [hunk 1]
```diff
@@ -1443,9 +1443,17 @@ private string GetParameterName(string prefix)
...
         }
 
-        private static void PopulateEagerLoadedNavigations(IncludeTreeNode includeTreeNode)
+        private void PopulateEagerLoadedNavigations(IncludeTreeNode includeTreeNode)
         {
             var entityType = includeTreeNode.EntityType;
+
+            if (_queryCompilationContext.IgnoreEagerLoadedNavigations
+                && !entityType.IsOwned()
+                && !entityType.HasDefiningNavigation())
```
Comment (AndriySvyryd) [inline src/EFCore/Query/Internal/NavigationExpandingExpressionVisitor.cs]: HasDefiningNavigation has nothing to do with this

File: src/EFCore/Query/Internal/NavigationExpandingExpressionVisitor.cs [hunk 2]
```diff
@@ -1443,9 +1443,17 @@ private string GetParameterName(string prefix)
             return uniqueName;
         }
 
-        private static void PopulateEagerLoadedNavigations(IncludeTreeNode includeTreeNode)
+        private void PopulateEagerLoadedNavigations(IncludeTreeNode includeTreeNode)
         {
             var entityType = includeTreeNode.EntityType;
+
+            if (_queryCompilationContext.IgnoreEagerLoadedNavigations
+                && !entityType.IsOwned()
```
Comment (AndriySvyryd) [inline src/EFCore/Query/Internal/NavigationExpandingExpressionVisitor.cs]: Only the ownership navigations should be excluded, other ones can still be ignored

File: src/EFCore/Query/Internal/NavigationExpandingExpressionVisitor.cs (hunk 1 above)
Comment (smitpatel) [reply src/EFCore/Query/Internal/NavigationExpandingExpressionVisitor.cs]: Does weak type have eager loaded navigations? And do we ignore it or not?

I copied these conditions based on what we expand in nav expansion.

Comment (AndriySvyryd) [reply src/EFCore/Query/Internal/NavigationExpandingExpressionVisitor.cs]: Non-owned weak entity types don't get eager loaded navigations by convention

**Comment by ajcvickers** [comment]
@smitpatel Remind me why this is marked as blocked?

**Comment by smitpatel** [comment]
We needed to add API to mark a navigation Eager loaded so that we can test this API.

**Pull request merged**